python/test/unit/language/test_matmul.py

```diff
@@ -475,6 +475,199 @@ def block_scale_mxfp_matmul(  #
+    b_ptrs = b_ptr + (offs_k_split[:, None] * stride_bk + offs_bn[None, :] * stride_bn)
+
+    # Create pointers for the first block of A and B scales
+    offs_asn = (pid_n * (BLOCK_N // 32) + tl.arange(0, (BLOCK_N // 32))) % N
+    offs_ks = tl.arange(0, BLOCK_K // SCALE_GROUP_SIZE * 32)
+
+    # B scales are N x K even though B operand is K x N.
+    b_scale_ptrs = (b_scales_ptr + offs_asn[:, None] * stride_bsn + offs_ks[None, :] * stride_bsk)
+    offs_asm = (pid_m * (BLOCK_M // 32) + tl.arange(0, (BLOCK_M // 32))) % M
+    a_scale_ptrs = (a_scales_ptr + offs_asm[:, None] * stride_asm + offs_ks[None, :] * stride_ask)
+    accumulator = tl.zeros((BLOCK_M, BLOCK_N), dtype=tl.float32)
+
+    for k in range(0, num_k_iter):
+        a_scales = tl.load(a_scale_ptrs).reshape(BLOCK_M // 32, BLOCK_K // 32 // 8, 4, 16, 2, 2,
+                                                 1).permute(0, 5, 3, 1, 4, 2, 6).reshape(BLOCK_M, BLOCK_K // 32)
+        b_scales = tl.load(b_scale_ptrs).reshape(BLOCK_N // 32, BLOCK_K // 32 // 8, 4, 16, 2, 2,
+                                                 1).permute(0, 5, 3, 1, 4, 2, 6).reshape(BLOCK_N, BLOCK_K // 32)
+        a_scales = tl.reshape(a_scales, (BLOCK_M, BLOCK_K // SCALE_GROUP_SIZE))
+        b_scales = tl.reshape(b_scales, (BLOCK_N, BLOCK_K // SCALE_GROUP_SIZE))
+
+        # Load the next block of A and B, generate a mask by checking the K dimension.
+        # If it is out of bounds, set it to 0.
+        if EVEN_K:
+            a = tl.load(a_ptrs)
+            b = tl.load(b_ptrs, cache_modifier=None)
+        else:
+            a = tl.load(a_ptrs, mask=offs_k[None, :] < K - k * (BLOCK_K // 2), other=0)
+            b = tl.load(b_ptrs, mask=offs_k[:, None] < K - k * (BLOCK_K // 2), other=0)
+
+        accumulator += tl.dot_scaled(a, a_scales, "e2m1", b, b_scales, "e2m1")
+
+        # Advance the ptrs to the next K block.
+        a_ptrs += (BLOCK_K // 2) * stride_ak
+        b_ptrs += (BLOCK_K // 2) * stride_bk
+        a_scale_ptrs += BLOCK_K * stride_ask
+        b_scale_ptrs += BLOCK_K * stride_bsk
+
+    c = accumulator.to(c_ptr.type.element_ty)
+
+    # Write back the block of the output matrix C with masks.
+    offs_cm = pid_m * BLOCK_M + tl.arange(0, BLOCK_M).to(tl.int64)
+    offs_cn = pid_n * BLOCK_N + tl.arange(0, BLOCK_N).to(tl.int64)
+    c_ptrs = (c_ptr + stride_cm * offs_cm[:, None] + stride_cn * offs_cn[None, :])
+    c_mask = (offs_cm[:, None] < M) & (offs_cn[None, :] < N)
+
+    tl.store(c_ptrs, c, mask=c_mask, cache_modifier=".wt")
+
+
+@pytest.mark.parametrize("M, N, K", [(1024, 1024, 1024), [512, 1024, 2048]])
+@pytest.mark.parametrize("BLOCK_M, BLOCK_N, BLOCK_K", [(256, 256, 256), (128, 128, 256)])
+@pytest.mark.skipif(is_cuda() or not is_hip_cdna4(), reason="AMD specific scale shuffling")
+@pytest.mark.skipif(not is_hip_cdna4(), reason="Requires hardware support for scaled mfma instructions")
+def test_preshuffle_scale_mxfp_amd(M, N, K, BLOCK_M, BLOCK_N, BLOCK_K, device):
```

Comment:
Nit: Instead of `_amd`, maybe name it as `_gfx950` to be more clear what architecture we are targeting.